Do not silently skip unknown OS

faucct opened the pull request:
I was trying to edit `~/.ssh/known_hosts` and because of this constraint action was just skipping without any logs even in debug mode. I think you should put these version constraints closer to the code they are related to.

joshcooper commented:
Hi @faucct. Thanks for your contribution. The module metadata doesn't affect any functionality on the server or agent. It's just used as guidance on the forge to indicate which platforms it should generally work on. What was the issue you ran into?

faucct commented:
I was running this resource with bolt and it was quietly skipping until I have added this module to puppetfile and changed its metadata. I will check if just adding the module is sufficient.

faucct commented:
Yes, for some reason just adding `mod 'puppetlabs-sshkeys_core'` to `Puppetfile` was sufficient. Could you tell me why? I am using bolt of version 1.29.0 on OS X distributed with brew.
